contracts/exchange-libs/contracts/src/LibTransactionDecoder.sol

```diff
@@ -0,0 +1,162 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.5;
+pragma experimental "ABIEncoderV2";
+
+import "@0x/contracts-exchange-libs/contracts/src/LibExchangeSelectors.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibOrder.sol";
+import "@0x/contracts-utils/contracts/src/LibBytes.sol";
+
+
+contract LibTransactionDecoder is LibExchangeSelectors {
+    using LibBytes for bytes;
+
+    function decodeTransaction(bytes memory signedTransactionData)
+        public
+        view
+        returns(
+            string memory functionName,
+            LibOrder.Order[] memory orders,
+            uint256[] memory takerAssetFillAmounts,
+            bytes[] memory signatures
+        )
+    {
+        bytes4 functionSelector = signedTransactionData.readBytes4(0);
+
+        if (functionSelector == BATCH_CANCEL_ORDERS_SELECTOR) {
+            functionName = "batchCancelOrders";
+        } else if (functionSelector == BATCH_FILL_ORDERS_SELECTOR) {
+            functionName = "batchFillOrders";
```

Comment:
I wonder if this giant if/else shouldn't be wrapped in a function in `LibExchangeSelectors`.